Tests/NIOSSLTests/NIOSSLIntegrationTest.swift

```diff
@@ -1038,6 +1038,36 @@ class NIOSSLIntegrationTest: XCTestCase {
             XCTFail("Encountered unexpected error: \(error)")
         }
     }
+    
+    func testUnprocessedDataOnReadPathBeforeClosing() throws {
+        let serverChannel = EmbeddedChannel()
+        let clientChannel = EmbeddedChannel()
+
+        let context = try configuredSSLContext()
+
+        try serverChannel.pipeline.addHandler(try NIOSSLServerHandler(context: context)).wait()
+        try clientChannel.pipeline.addHandler(try NIOSSLClientHandler(context: context, serverHostname: nil)).wait()
+
+        let addr = try SocketAddress(unixDomainSocketPath: "/tmp/whatever2")
+        let connectFuture = clientChannel.connect(to: addr)
+        serverChannel.pipeline.fireChannelActive()
+        try interactInMemory(clientChannel: clientChannel, serverChannel: serverChannel)
+        try connectFuture.wait()
+
+        // Ok, we're connected. Now we want to close the server, and have that trigger a client CLOSE_NOTIFY.
+        // After the CLOSE_NOTIFY create another chunk of data.
+        let _ = serverChannel.close()
+        
+        // Create a new chunk of data after the close.
+        var clientBuffer = clientChannel.allocator.buffer(capacity: 5)
+        clientBuffer.writeStaticString("hello")
+        _ = try clientChannel.writeAndFlush(clientBuffer).wait()
+        let clientClosePromise = clientChannel.close()
```

Comment:
Great, let's return to the feedback I gave here. I'd like us to confirm that the server channel actually did receive the client data, _and_ that the channel is now closed. Can we update the test to do that?

Reply:
Feedback addressed and tested on macOS and Linux.